Fb365

joelaro04 opened the pull request.

joelaro04 committed:
Merge branch 'master' of github.com:spiretrading/pyaugur

joelaro04 committed:
Implemented base AugurClient class.

joelaro04 committed:
Implementation for load_market_info method

joelaro04 committed:
Cast MarketInfo parameters to Decimal and datetime

joelaro04 committed:
Added addresses.json

joelaro04 committed:
Added exception handler for AugurClient

joelaro04 committed:
Merged with master

joelaro04 committed:
Translated camel case from node to snake case

joelaro04 committed:
Removed requests import

joelaro04 committed:
Removed unused variables

joelaro04 committed:
Cast reporting_state from node to a ReportingState enum

joelaro04 committed:
Dynamically cast types for MarketInfo objects

joelaro04 committed:
Reset MarketInfo

joelaro04 committed:
MarketInfo.tags does not return None. It can return a list of None.

joelaro04 committed:
Cast OutcomeInfo and added validation for empty list/strings

joelaro04 committed:
Omit all None types in lists.

- Fixed datetime.fromtimestamp from raising and error when passed
microseconds.

joelaro04 committed:
Removed MarketInfo.tags returning a list of None.

joelaro04 committed:
Properties that expect a list or str will no longer return None

joelaro04 committed:
Added empty list/default for OutcomeInfo and NormalizedPayout in AugurClient.

joelaro04 committed:
Removed shortnames

joelaro04 committed:
Cast Augur types

joelaro04 committed:
Changed list() to []. Added exception for timestamps.

joelaro04 committed:
Removed get_markets for testing

joelaro04 committed:
Update comments.

joelaro04 committed:
Refactor load_market_info

joelaro04 committed:
Added methods for ensuring data type on MarketInfo creation.

kranar commented on augur/augur_client.py:
```diff
@@ -0,0 +1,168 @@
+
+  def close(self):
+    '''Disconnects from an Augur node.'''
+    self._is_open = False
```
You will need to perform an actual disconnection here.

kranar commented on augur/augur_client.py:
```diff
@@ -0,0 +1,168 @@
+      raise IOError(response_error)
+
+  def _to_snake_case(self, text):
+    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', text)
```
Make this an internal free function rather than a method.

joelaro04 replied:
I realize this function doesn't get used in the current implementation. I'll remove this for now.

kranar commented on augur/augur_client.py:
```diff
@@ -0,0 +1,168 @@
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
+  def _ensure_unix_timestamp(self, timestamp):
+    try:
```
Make this an internal free function instead of a method. In general any functions that do not depend on ```self``` should be free functions. Internal functions are prefixed with an underscore.

kranar reviewed (changes requested):
Much much better.

joelaro04 committed:
Websocket can now be disconnected.

- Methods that did not rely on class variables are now functions
outside the class.

joelaro04 committed:
Renamed function parameters to better describe what is being passed in.
